pkg/kfilefields/bpf/filefields.bpf.c

```diff
@@ -66,6 +68,8 @@ int BPF_KRETPROBE(ig_fget_x, struct file *ret)
 
 	ff->private_data = (u64)BPF_CORE_READ(ret, private_data);
 	ff->f_op = (u64)BPF_CORE_READ(ret, f_op);
+	private_file = (struct file *)ff->private_data;
+	ff->private_inode = (u64)BPF_CORE_READ(private_file, f_inode);
```

Comment:
`(struct file *)->private_data` is casted to a `struct file *` but this is only valid if the filesystem is overlay.

We can check if it is the case with `(struct file *)->i_sb->s_type->name == "overlay"`.